### PR TITLE
feat(TBD-13632): Dataproc GA

### PR DESCRIPTION
**Pull request by sponomarova**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveConnection/tHiveConnection_java.xml [hunk 1]
```diff
@@ -63,13 +56,13 @@
 
 		<PARAMETER NAME="CONNECTION_MODE" FIELD="CLOSED_LIST"
 			NUM_ROW="15" REQUIRED="true" REPOSITORY_VALUE="CONNECTION_MODE"
-			SHOW_IF="!DISTRIB[DISTRIBUTION, HIVE_VERSION].useCloudLauncher[]"
+			SHOW_IF="(!DISTRIB[DISTRIBUTION, HIVE_VERSION].useCloudLauncher[] AND !DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportUniversalDataprocMode[])">
 			GROUP="CONNECTION">
 			<ITEMS DEFAULT="EMBEDDED">
 				<ITEM NAME="EMBEDDED" VALUE="EMBEDDED"
-					SHOW_IF="(DISTRIBUTION=='CUSTOM') OR DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportEmbeddedMode[]" />
+					SHOW_IF="isShowp[CONNECTION_MODE] AND (DISTRIBUTION=='CUSTOM' OR !DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportStandaloneMode[])" />
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveConnection/tHiveConnection_java.xml:63]: typo "isShowp"

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveConnection/tHiveConnection_java.xml [hunk 2]
```diff
@@ -63,13 +56,13 @@
...
 				<ITEM NAME="STANDALONE" VALUE="STANDALONE"
-					SHOW_IF="(DISTRIBUTION=='CUSTOM') OR (HIVE_SERVER == 'HIVE2' AND DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportStandaloneMode[]) OR (HIVE_SERVER == 'HIVE' AND DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportHive1Standalone[])" />
+					SHOW_IF="isShowp[CONNECTION_MODE] AND (DISTRIBUTION=='CUSTOM') OR (HIVE_SERVER == 'HIVE2' AND DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportStandaloneMode[]) OR (HIVE_SERVER == 'HIVE' AND DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportHive1Standalone[])" />
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveConnection/tHiveConnection_java.xml:65]: typo "isShowp"

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveIn/tHiveIn_java.xml [hunk 3]
```diff
@@ -209,6 +215,9 @@
             <IMPORT NAME="azure-storage-1.2.0" MODULE="azure-storage-1.2.0.jar" MVN="mvn:org.talend.libraries/azure-storage-1.2.0/6.0.0" 
                 REQUIRED_IF="(USE_EXISTING_CONNECTION=='false') AND (DISTRIBUTION=='MICROSOFT_HD_INSIGHT') AND (DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportOldImportMode[])" />
 			<!-- #################### -->
+			<!-- Dataproc 2.x -->
+			<IMPORT MODULE_GROUP="DATAPROC-LAUNCHER" 
+                    REQUIRED_IF="(USE_EXISTING_CONNECTION=='false) AND (DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportUniversalDataprocMode[] AND DISTRIBUTION!='CUSTOM')" REQUIRED="true"/>
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveIn/tHiveIn_java.xml:220]: should not you add this module group to other hive components as well ?

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveIn/tHiveIn_java.xml [hunk 4]
```diff
@@ -82,28 +82,31 @@
...
+		</PARAMETER>
+		<PARAMETER NAME="GOOGLE_REGION" REPOSITORY_VALUE="GOOGLE_REGION" FIELD="TEXT" REQUIRED="true"
+			NUM_ROW="18" GROUP="CONN"
+			SHOW_IF="DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportUniversalDataprocMode[]">
+			<DEFAULT>"global"</DEFAULT>
+		</PARAMETER>
+		<PARAMETER NAME="GOOGLE_JARS_BUCKET" REPOSITORY_VALUE="GOOGLE_JARS_BUCKET" FIELD="TEXT" REQUIRED="true"
+			NUM_ROW="19" GROUP="CONN"
+			SHOW_IF="DISTRIB[DISTRIBUTION, HIVE_VERSION].doSupportUniversalDataprocMode[]">
+			<DEFAULT>"gs://my-bucket/talend/jars"</DEFAULT>
```
Comment (lbourgeois) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveIn/tHiveIn_java.xml:104]: Bad thing is that those will be displayed as soon as Univ/311 will be choosen even if user does not plan to use Dataproc. That's why a new entry Dataproc was added to Connection mode for tHiveConnection. I think same should be done for other components.

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveConnection/tHiveConnection_begin.javajet [hunk 5]
```diff
@@ -27,7 +27,7 @@ imports="
 %>
         <%@ include file="@{org.talend.designer.components.localprovider}/components/tHiveConnection/tHiveConnection_azure.javajet"%>
 <%
-    } else if (hiveDistrib.isGoogleDataprocDistribution()) { // Connection through Dataproc api
+    } else if (hiveDistrib.doSupportUniversalDataprocMode()) { // Connection through Dataproc api
```
Comment (lbourgeois) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveConnection/tHiveConnection_begin.javajet:30]: I think we have an issue if user wants to use Univ/311/Hive standalone and not dataproc

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveIn/tHiveIn_java.xml (hunk 4 above)
Comment (sponomarova) [reply main/plugins/org.talend.designer.components.localprovider/components/tHiveIn/tHiveIn_java.xml:104]: I will add  the same to other Hive components.

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet [hunk 6]
```diff
@@ -71,29 +71,29 @@ if(hiveDistrib.isExecutedThroughWebHCat()) {
...
-	}
+            }
 <%
-} else if (hiveDistrib.isGoogleDataprocDistribution()) {
-    if(isLog4jEnabled) {
+        } else if (theVersion.equals("DATAPROC")) { // dataproc
+            if(isLog4jEnabled) {
 %>
-        log.debug("Query for <%=cid%>: " + query_<%=cid %>.replace("';'", "'\\;'") + ";");
+                log.debug("Query for <%=cid%>: " + querySQL_<%=cid %>.replace("';'", "'\\;'") + ";");
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet:94]: shouldn't it be query_ ?

Comment (sponomarova) [reply main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_main.javajet:94]: yes, I will correct.

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet [hunk 7]
```diff
@@ -89,12 +88,12 @@ imports="
...
 <%
 		}
-    } else if(hiveDistrib.isGoogleDataprocDistribution()) {
+    } else if (theVersion.equals("DATAPROC")) {
 		// Distribution: Google Dataproc
 %>
         <%@ include file="@{org.talend.designer.components.localprovider}/components/templates/Hive/GetDataprocConnection.javajet"%>
 <%
-	} else {
+	} else if (theVersion.equals("STANDALONE")) {
```
Comment (AlixMetivier) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet:96]: why does else becomes else if standalone ? is that really the only case ? shouldn't you keep else as it is ?

**Review by AlixMetivier — APPROVED** [review]

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet (hunk 7 above)
Comment (sponomarova) [reply main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet:96]: yes, right. Will change.

**Pull request closed without merging**